app/components/Footnote.tsx

```diff
@@ -8,10 +8,17 @@ interface Props {
 }
 
 export default function Footnote({ number, children }: Props) {
-  const { backgroundColor, primaryTextColor } = useContext(ChapterContext);
+  const { backgroundColor, primaryTextColor, footnoteState, setFootnoteState, } = useContext(ChapterContext);
   const [isTextVisible, setIsTextVisible] = useState(false);
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    setFootnoteState(footnoteState + 1);
```

Comment:
React setters can take the previous state:


```suggestion
    setFootnoteState(footnoteState => footnoteState + 1);
```

Reply:
Good to know.